src/no-side-effects.spec.ts

```diff
@@ -176,37 +294,125 @@ createTester().run(ruleId, rule, {
+
+              Object.freeze(object);
+              
+              try {
+                 object.prop = 'bar';
+              } catch (error) {
+                  console.error(error); 
+              }
+              
+              Symbol.for('foo');
+              export default obj;
+              const server = http.createServer(logger(configuration.requestHandler, unresolvedConfiguration.name));
+              server.listen(Number.parseInt(unresolvedConfiguration.env['PORT'], 10), '0.0.0.0');
+              
+              await new Promise((resolve) => {
+                server.on('listening', resolve);
+              });
+              export * from './module';`,
       errors: [
         { messageId: 'NO_SIDE_EFFECTS' },
         { messageId: 'NO_SIDE_EFFECTS' },
         { messageId: 'NO_SIDE_EFFECTS' },
         { messageId: 'NO_SIDE_EFFECTS' },
+      ],
+      options: [{ excludedIdentifiers: ['debug', 'log', 'Symbol.for', 'Object.freeze'] }],
+      name: 'Invalid case with multiple side effects including export all declarations and async operations with try catch',
+    },
+    {
+      code: `const x = 10;
+        let y = 20;
+        var x = 10;
+        if (x > y) {
+          y = x;
+        } else {
+          y = 30;
+        }
+        switch (y) {
+          case 10:
+            break;
+          default:
+            return;
+        }
+       
+        class MyClass {
+          constructor() {}
+          static myStaticMethod() {}
+          private myPrivateMethod() {}
+        }
+       
+        enum MyEnum {
+          A,
+          B,
+          C
+        }
+        namespace MyNamespace {
+          export const value = 42;
+        }
+        import { something } from 'somewhere';
+        export const myExport = 100;
+        const myVar: any = null;
+        const myBool: boolean = true;
+        const myNum: number = 123;
+        const myStr: string = 'hello';
+        const myVoid: void = undefined;
+        const myNull: null = null;
+        const myUndefined: undefined = undefined;
+        const myNever: never = (() => { throw new Error('never'); })();
+        const mySymbol: symbol = Symbol('mySymbol');
+        const myObject: object = {};
+        const myUnknown: unknown = 'unknown';
+        const myTrue: true = true;
+        const myFalse: false = false;
+        function genericFunction<T>(arg: T): T {
+          return arg;
+        }
+        module MyModule {
+          export const moduleValue = 50;
+        }
+        const isTypeof = typeof x === 'number';
+        const isInstanceof = y instanceof Number;
+        const isIn = 'prop' in myObject;
+        const isNew = new MyClass();
+        this.myMethod();
+        declare const myDeclare: string;
+        async function myAsyncFunction() {
+          await Promise.resolve();
+        }
+        function* myGenerator() {
+          yield 1;
+        }
+        using myResource = { dispose() {} };`,
+      errors: [
```

Comment:
with more side-effect scenarios, it's becoming more complex.
is that possible to report some message data to describe which type of side-effect it vialates?
it'll not only help understand the reason, but also make the tests more specific especially when multiple errors are combined in the same test case.